p2p: Return a max of 1000 addrs from CAddrMan::GetAddr

Empact opened the pull request:
This is the documented max size for an addr response, and was separately being
limited in PushAddress to 1000 via MAX_ADDR_TO_SEND. No point in selecting a
too-large random set then filtering it randomly down to a smaller set.

Empact committed:
p2p: Return a max of 1000 addrs from CAddrMan::GetAddr

This is the documented max size for an addr response, and was separately being
limited in PushAddress to 1000 via MAX_ADDR_TO_SEND. No point in selecting a
too-large random set then filtering it randomly down to a smaller set.

MarcoFalke commented:
I think addrman would return a larger list that is node-agnostic and then net would filter that list taking into account the node state, no?

practicalswift commented on src/addrman.h:
```diff
@@ -172,7 +172,7 @@ class CAddrInfo : public CAddress
 
 //! the maximum number of nodes to return in a getaddr call
-#define ADDRMAN_GETADDR_MAX 2500
+#define ADDRMAN_GETADDR_MAX 1000
```
When we're editing this file – what about skipping `#define` and using proper constants instead? :-)

gmaxwell commented:
I thought the two layer setup was so that multiple probes would get more addresses but not enumerate the whole table.

gmaxwell commented:
@sipa ^

laanwj commented on src/net_processing.cpp:
```diff
@@ -1757,7 +1757,7 @@ bool static ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStr
 
             // Get recent addresses
-            if (pfrom->fOneShot || pfrom->nVersion >= CADDR_TIME_VERSION || connman->GetAddressCount() < 1000)
+            if (pfrom->fOneShot || pfrom->nVersion >= CADDR_TIME_VERSION || connman->GetAddressCount() < ADDRMAN_GETADDR_MAX)
```
I like changing these to a constant, however "maximum number of addresses in GETADDR" is a P2P layer constant, I don't think it should use addrman's internal constant.

MarcoFalke commented:
@Empact `PushAddr` will return early when `addrKnown.contains` it. So when `vAddr` is of size 1000, it might very well send less than 1000 addresses back in the response, even though there are plenty to send. No?

Empact commented:
@MarcoFalke yep that sounds like the explanation. Thanks!

Empact committed:
Don't return known addresses from CConnman::GetAddresses

These addresses are used expressly for sending to a node in response to
GETADDR, so we may as well skip them over in the lookup soas to return the
appropriate number.
